cypress/e2e/delegation.cy.ts

```diff
@@ -5,7 +5,7 @@ describe('delegation', () => {
     cy.resetBlockchain().login();
 
     // Approve, deposit and stake
-    cy.findByText('+ Deposit').click();
+    cy.dataCy('staking-cards').findByRole('button', { name: 'Deposit' }).click();
```

Comment:
Lets rather use the built in `data-testid="..."` and `cy.findByTestId(...)` for all of these `cy.dataCy()` usages that you've added.

Reply:
Done